src/python_testing/TC_CCTRL_2_3.py

```diff
@@ -0,0 +1,179 @@
+        events = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=event_path)
+        asserts.assert_equal(len(events), 1, "Unexpected event list len")
+        asserts.assert_equal(events[0].Data.statusCode, 0, "Unexpected status code")
+        asserts.assert_equal(events[0].Data.clientNodeId,
+                             self.matter_test_config.controller_node_id, "Unexpected client node id")
+        asserts.assert_equal(events[0].Data.requestId, good_request_id, "Unexpected request ID")
+
+        self.step(7)
+        cmd = Clusters.CommissionerControl.Commands.RequestCommissioningApproval(
+            requestId=good_request_id, vendorId=th_server_vid, productId=th_server_pid)
+        try:
+            await self.send_single_cmd(cmd=cmd)
+            asserts.fail("Unexpected success on CommissionNode")
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Failure, "Incorrect error returned")
+
+        self.step(8)
+        cmd = Clusters.CommissionerControl.Commands.CommissionNode(requestId=good_request_id, responseTimeoutSeconds=30)
+        resp = await self.send_single_cmd(cmd)
+        asserts.assert_equal(type(resp), Clusters.CommissionerControl.Commands.ReverseOpenCommissioningWindow,
+                             "Incorrect response type")
+
+        self.step(9)
+        cmd = Clusters.CommissionerControl.Commands.CommissionNode(requestId=good_request_id, responseTimeoutSeconds=30)
+        try:
+            await self.send_single_cmd(cmd=cmd)
+            asserts.fail("Unexpected success on CommissionNode")
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Failure, "Incorrect error returned")
+
+        self.step(10)
+        # min commissioning timeout is 3*60 seconds, so use that even though the command said 30.
+        cmd = Clusters.AdministratorCommissioning.Commands.OpenCommissioningWindow(commissioningTimeout=3*60,
+                                                                                   PAKEPasscodeVerifier=resp.PAKEPasscodeVerifier,
+                                                                                   discriminator=resp.discriminator,
+                                                                                   iterations=resp.iterations, salt=resp.salt)
+        await self.send_single_cmd(cmd, dev_ctrl=self.TH_server_controller, node_id=self.server_nodeid, endpoint=0, timedRequestTimeoutMs=5000)
+
+        self.step(11)
+        if not self.is_ci:
+            time.sleep(30)
+
+        self.step(12)
+        th_server_fabrics_new = await self.read_single_attribute_check_success(cluster=Clusters.OperationalCredentials, attribute=Clusters.OperationalCredentials.Attributes.Fabrics, dev_ctrl=self.TH_server_controller, node_id=self.server_nodeid, endpoint=0, fabric_filtered=False)
+        # TODO: this should be mocked too.
```

Comment:
This TODO comment is not relavent for you. It is valid in CCTRL_2_2 because we had `src/python_testing/test_testing/test_TC_CCNTL_2_2.py` to test the test